core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2105,6 +2105,12 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
             traceDBIOWithParent("getPolicySyncStatus", s1)(_ => DBIO.from(samDAO.getPolicySyncStatus(SamResourceTypeNames.billingProject, workspaceRequest.namespace, SamBillingProjectPolicyNames.owner, userInfo).map(_.email))).flatMap { projectOwnerPolicyEmail =>
               traceDBIOWithParent("setupGoogleProject", s1)(_ => DBIO.from(setupGoogleProject(billingProject, billingAccount, s1, workspaceId))).flatMap { case (googleProjectId, googleProjectNumber) =>
                 traceDBIOWithParent("saveNewWorkspace", s1)(s2 => saveNewWorkspace(workspaceId, workspaceRequest, bucketName, projectOwnerPolicyEmail, googleProjectId, Option(googleProjectNumber), Option(billingAccount), dataAccess, s2).flatMap { case (savedWorkspace, policyMap) =>
+                  // After the workspace has been created, create the google-project resource in Sam with the workspace as the resource parent
+                  val googleProjectOwnerPolicy = SamGoogleProjectPolicyNames.owner -> SamPolicy(Set.empty, Set.empty, Set(SamGoogleProjectRoles.owner))
+                  val googleProjectNotebookUserPolicy = SamGoogleProjectPolicyNames.notebookUser -> SamPolicy(Set.empty, Set.empty, Set(SamGoogleProjectRoles.notebookUser))
+                  val googleProjectPolicies = Map(googleProjectOwnerPolicy, googleProjectNotebookUserPolicy)
+                  traceDBIOWithParent("createResourceFull (google project)", s1)(_ => DBIO.from(samDAO.createResourceFull(SamResourceTypeNames.googleProject, googleProjectId.value, googleProjectPolicies, workspaceRequest.authorizationDomain.getOrElse(Set.empty).map(_.membersGroupName.value), userInfo, Option(SamFullyQualifiedResourceId(workspaceId, SamResourceTypeNames.workspace.value)))))
+
```

Comment:
this is the meat of the PR. I'm open to moving it elsewhere (or maybe into an inner function) if folks have suggestions.